From 8b1bda73dd16fffd7285bb19deedf355cd9737ee Mon Sep 17 00:00:00 2001
From: Tiger Liu <liuyihu@kylinos.cn>
Date: Fri, 1 May 2026 12:59:10 +0800
Subject: [PATCH 11/20] =?UTF-8?q?fix:=C2=A0fix=20the=20lagging=20issue=20d?=
 =?UTF-8?q?uring=20mouse=20movement?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

For glenfly 10c0t gpu, there's a lagging issue during mouse movement
when performing large language model (LLM) inference.

Bug: #382109

From: kylin

Severity: Critical

Changelog-Bug: #382109 【AI助手】【文本对话】在回答过程中鼠标滑动卡顿
Change-Id: Iaceef0e0143b8f1bf2542a746b80dc27a8132792
---
 ggml/src/ggml-ponn/ggml-ponn.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ggml/src/ggml-ponn/ggml-ponn.cpp b/ggml/src/ggml-ponn/ggml-ponn.cpp
index f789389..d14e587 100644
--- a/ggml/src/ggml-ponn/ggml-ponn.cpp
+++ b/ggml/src/ggml-ponn/ggml-ponn.cpp
@@ -1263,6 +1263,11 @@ static enum ggml_status ggml_backend_ponn_graph_compute(
             return GGML_STATUS_ABORTED;
         }
 
+	//目前硬编码为cgraph每1/6 Sync一次，其中6=(115/20)+1
+	if(i % (cgraph->n_nodes / 6) == 0) {
+	    ponnSyncStream(ponnGetStream());
+	}
+
         if (ggml_is_empty(node) || node->op == GGML_OP_NONE || i == fused_index) {
             continue;
         }
-- 
2.43.0

